homeassistant/components/hydrawise/sensor.py

```diff
@@ -44,23 +48,21 @@ def update(self):
         """Get the latest data and updates the states."""
         mydata = self.hass.data[DATA_HYDRAWISE].data
         _LOGGER.debug("Updating Hydrawise sensor: %s", self._name)
+        relay_data = mydata.relays[self.data["relay"] - 1]
         if self._sensor_type == "watering_time":
-            if not mydata.running:
-                self._state = 0
+            if relay_data["timestr"] == "Now":
+                self._state = int(relay_data["run"] / 60)
             else:
-                if int(mydata.running[0]["relay"]) == self.data["relay"]:
-                    self._state = int(mydata.running[0]["time_left"] / 60)
-                else:
-                    self._state = 0
+                self._state = 0
         else:  # _sensor_type == 'next_cycle'
-            for relay in mydata.relays:
-                if relay["relay"] == self.data["relay"]:
-                    if relay["nicetime"] == "Not scheduled":
-                        self._state = "not_scheduled"
-                    else:
-                        self._state = f"{relay['nicetime'].split(',')[0]} {relay['nicetime'].split(' ')[3]}"
+            next_cycle = min(relay_data["time"], TWO_YEAR_SECONDS)
+            _LOGGER.debug("New cycle time: %s", next_cycle)
+            self._state = dt.as_local(
+                dt.utc_from_timestamp(dt.as_timestamp(dt.now()) + next_cycle)
+            ).isoformat()
```

Comment:
The time should be UTC time.
```suggestion
            self._state = dt.utc_from_timestamp(time.time() + next_cycle).isoformat()
```

Reply:
If I don't use `dt.as_local()` then the time is incorrect. It shows UTC, not EST. My configuration.yaml file has this in it:
```
homeassistant:
  time_zone: America/New_York
```
But even with this I'm getting a time value that's 4 hours off. If I add `dt.as_local()` then the time is accurate to my timezone. 

Should I still show UTC anyway?

Reply:
I removed `as_local()`. I'm assuming that the correct local time will be displayed in the released version. I'm not understanding this clearly.

Reply:
All times in the state machine should be UTC. This is our standard.

In the frontend the user can configure this sensor timestamp to be shown in different time formats:
https://www.home-assistant.io/lovelace/entities/#format